cmd/ingest/ingest.go

```diff
@@ -42,24 +57,170 @@ func main() {
 	var message Message
 
 	go func() {
-		for d := range broker.GetMessages(mq, config.Broker.Queue) {
-			log.Debugf("received a message: %s", d.Body)
-			err := json.Unmarshal(bytes.Replace(d.Body, []byte(`'`), []byte(`"`), -1), &message)
-			if err != nil {
+		for delivered := range broker.GetMessages(mq, config.Broker.Queue) {
+			log.Debugf("Received a message: %s", delivered.Body)
+			if err := json.Unmarshal(delivered.Body, &message); err != nil {
 				log.Errorf("Not a json message: %s", err)
 			}
 
-			e := reflect.ValueOf(&message).Elem()
-			for i := 0; i < e.NumField(); i++ {
-				if e.Field(i).Interface() == "" || e.Field(i).Interface() == 0 {
-					log.Errorf("%s is missing", e.Type().Field(i).Name)
-					err = fmt.Errorf("%s is missing", e.Type().Field(i).Name)
+			fileID, err := db.InsertFile(fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath), message.User)
+			if err != nil {
+				log.Errorf("InsertFile failed, reason: %v", err)
+				// This should really be hadled by the DB retry mechanism
+			}
+
+			file, err := storage.ReadFile(config.Inbox.Type, fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath))
+			if err != nil {
+				log.Errorf("Failed to open file: %s, reason: %v", fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath), err)
+				continue
+			}
+
+			fileSize, err := storage.GetFileSize(config.Inbox.Type, fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath))
+			if err != nil {
+				log.Errorf("Failed to get file size: %s, reason: %v", fmt.Sprintf("%s/%s", config.Archive.Location, message.FilePath), err)
+				continue
+			}
+
+			// 4MiB readbuffer, this must be large enough that we get the entire header and the first 64KiB datablock 
+			// Should be made configurable once we have S3 support
+			readBuffer := make([]byte, 4 * 1024 * 1024)
```

Comment:
This should be fixed

Reply:
What should be fixed?

Reply:
The size of the bytes array should be configurable (from the comment above)